src/igraph/drawing/graph.py

```diff
@@ -1010,6 +1012,61 @@ def callback_edge_offset(event):
 
         # FIXME: deal with unnamed *args
 
+        # graph is not necessarily a graph, it can be a VertexClustering. If so
+        # extract the graph. The clustering itself can be overridden using
+        # the "mark_groups" option
+        clustering = None
+        if isinstance(graph, (VertexClustering, VertexCover)):
+            clustering = graph
+            graph = clustering.graph
+
+        # mark groups: the used data structure is eventually the dict option:
+        # tuples of vertex indices as the keys, colors as the values. We
+        # convert other formats into that one here
+        if "mark_groups" in kwds:
+            if kwds["mark_groups"] is False:
+                del kwds["mark_groups"]
+            elif (kwds["mark_groups"] is True) and (clustering is not None):
+                pass
+            # Deferred import to avoid a cycle in the import graph
+            elif isinstance(kwds["mark_groups"], (VertexClustering, VertexCover)):
+                if clustering is not None:
+                    raise ValueError(
+                        "mark_groups cannot override a clustering with another"
+                    )
+                else:
+                    clustering = kwds["mark_groups"]
+                    kwds["mark_groups"] = True
+            else:
+                try:
+                    mg_iter = iter(kwds["mark_groups"])
+                except TypeError:
+                    raise TypeError("mark_groups is not in the right format")
+                kwds["mark_groups"] = dict(mg_iter)
+
+        # If a clustering is set, color vertices and mark groups if requested
+        if clustering is not None:
+            if "vertex_color" not in kwds:
+                membership = clustering.membership
+                if isinstance(clustering, VertexCover):
+                    membership = [x[0] for x in membership]
+                clusters = sorted(set(membership))
+                n_colors = len(clusters)
+                cmap = mpl.cm.get_cmap("viridis")
+                colors = [cmap(1.0 * i / n_colors) for i in range(n_colors)]
+                c = [colors[clusters.index(i)] for i in membership]
+                kwds["vertex_color"] = c
+
+                # mark_groups if not explicitely marked
+                if ("mark_groups" in kwds) and (kwds["mark_groups"] is True):
```

Comment:
`if kwds.get("mark_groups") is True` is enough; `.get()` returns `None` if the key is not in the dict